### PR TITLE
Change "Server" to "Enterprise"

### PR DESCRIPTION
**Pull request by royb-legit**
Following some feedback about the fact that the "Server" section creates confusion - we should change the term to the more general "Enterprise" namespace. That's because the policies under this section don't apply just to servers (GHES / GitLab Self-Managed), They also apply for cloud-based SCMs (e.g. [GitHub Enterprise](https://github.com/enterprise)).
Also in this PR: 
- Additional guide targets
- Putting Legitify as the first suggested tool, as it's the only one that covers all of the guide's policies

### CODE REVIEW

**Review by caabernathy — APPROVED** [review]
This looks good to me. Thanks for the additional fixes around contributors and adding a new audience persona.

File: docs/SCM-BestPractices/README.md [hunk 1]
```diff
@@ -10,21 +10,16 @@ Collaborative source code management platforms (such as GitHub and GitLab) play
 
 This guide has been written for the:
 
-* **Maintainer** who wants to improve the security posture for one or more GitHub repositories or GitLab projects they support.  
+* **Maintainer** who wants to improve the security posture for one or more GitHub repositories or GitLab projects they support.
+* **Owner** who wants to improve the security posture for their GitHub organization or GitLab group they manage. 
 * **Open Source Program Office (OSPO)** (or a team that plays a similar role) who is typically responsible for multiple GitHub organizations or GitLab groups.
 * **Operations** team tasked with applying policies as part of their work managing assets on these platforms.
+* **GitHub/GitLab server administrator** who wants to improve the security posture for their SCM server.
```
Comment (ctcpip) [inline docs/SCM-BestPractices/README.md:17]: this doesn't seem appropriate for this guide.  (server security)

Comment (ctcpip) [inline docs/SCM-BestPractices/README.md:17]:
```suggestion
* **GitHub/GitLab enterprise administrator** who wants to improve the security posture for their SCM enterprise. 
```

File: docs/SCM-BestPractices/gitlab/README.md [hunk 2]
```diff
@@ -17,14 +17,17 @@ This guide has been written for the:
 
 * **Maintainer** who wants to improve the security posture for one or more
 GitLab projects they support.
+* **Owner** who wants to improve the security posture for one or more GitLab
+groups they manage. 
 * **Open Source Program Office (OSPO)** who is typically responsible for
 multiple groups and projects.
 * **Operations** team tasked with applying policies as part of their work
 managing assets on GitLab.
+* **GitLab server admin** who wants to improve the security posture for their server.
```
Comment (ctcpip) [inline docs/SCM-BestPractices/gitlab/README.md:26]: need to find out what GitLab calls this.  is it server admin?  is enterprise admin?  something else?

File: docs/SCM-BestPractices/README.md [hunk 3]
```diff
@@ -10,21 +10,16 @@ Collaborative source code management platforms (such as GitHub and GitLab) play
...
+* **Owner** who wants to improve the security posture for their GitHub organization or GitLab group they manage. 
 * **Open Source Program Office (OSPO)** (or a team that plays a similar role) who is typically responsible for multiple GitHub organizations or GitLab groups.
 * **Operations** team tasked with applying policies as part of their work managing assets on these platforms.
+* **GitHub/GitLab server administrator** who wants to improve the security posture for their SCM server. 
 
 ## Tooling
 
 Below is a non-exhaustive list of possible tools that can be used to assist in review source code repositories.
 
-### Allstar - <https://github.com/ossf/allstar>
```
Comment (ctcpip) [inline docs/SCM-BestPractices/README.md:21]: we should keep this section alphabetical

File: docs/SCM-BestPractices/README.md [hunk 4]
```diff
@@ -161,5 +163,5 @@ The following community members helped contribute to this guidance:
 * [Chris de Almeida, IBM](https://github.com/ctcpip)
 * [Christine Abernathy, F5 - project lead](https://github.com/caabernathy)
 * [Daniel Appelquist, Snyk - project lead](https://github.com/Torgo)
-* [Noam Dotan, Legit Security - project lead](https://github.com/)
+* [Noam Dotan, Legit Security - project lead](https://github.com/noamd-legit)
```
Comment (ctcpip) [inline docs/SCM-BestPractices/README.md:166]: this is fine but really should be in a separate PR

File: docs/SCM-BestPractices/github/README.md [hunk 5]
```diff
@@ -17,10 +17,14 @@ This guide has been written for the:
...
 * **Maintainer** who wants to improve the security posture for one or more
 GitHub repositories they support.
+* **Owner** who wants to improve the security posture for one or more GitHub
+organizations they manage. 
 * **Open Source Program Office (OSPO)** who is typically responsible for
 multiple organizations and repositories.
 * **Operations** team tasked with applying policies as part of their work
 managing assets on GitHub.
+* **GitHub Enterprise Server admin** who wants to improve the security posture
+for their server.
```
Comment (ctcpip) [inline docs/SCM-BestPractices/github/README.md:27]:
```suggestion
* **GitHub Enterprise administrator** who wants to improve the security posture for their enterprise.
```

File: docs/SCM-BestPractices/README.md (hunk 3 above)
Comment (royb-legit) [reply docs/SCM-BestPractices/README.md:21]: Legitify should be the first suggested tool, as it's the only one that covers all of the guide's policies

File: docs/SCM-BestPractices/README.md (hunk 4 above)
Comment (royb-legit) [reply docs/SCM-BestPractices/README.md:166]: Yeah, just didn't think it's worth it for this additional small fix
